refactor: remove redundant pylint disable from celery instrumentation

forafox opened the pull request:
# Description

This PR removes the redundant `pylint: disable=attribute-defined-outside-init` comment from the celery instrumentation and adds this rule to the global pylint disable list in `.pylintrc`. 

The `attribute-defined-outside-init` pylint rule is problematic for OpenTelemetry instrumentations because attributes are commonly set during instrumentation (in methods like `_instrument()`) rather than in `__init__`. This pattern is used throughout the codebase (30+ occurrences) and is a standard practice for OpenTelemetry instrumentations where attributes are initialized during the instrumentation process.

By disabling this rule globally, we eliminate the need for individual `pylint: disable` comments across the codebase, making the code cleaner and more maintainable.

Fixes #3828

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [x] Verified that the celery instrumentation file compiles without syntax errors
- [x] Confirmed that the pylint disable comment has been completely removed from celery instrumentation
- [x] Verified that `attribute-defined-outside-init` has been added to the global pylint disable list
- [x] Ran pylint checks to ensure no new warnings are introduced

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

forafox committed:
refactor: remove redundant pylint disable from celery instrumentation

xrmx reviewed (commented):
If we are going to move this globally I would expect the other 30 occurences of the pylint disable to move removed too. Said that the lint warning may be useful in other cases.

forafox committed:
cleanup: remove redundant pylint disable comments for attribute-defined-outside-init

forafox committed:
cleanup: remove redundant pylint disable comments for attribute-defined-outside-init

forafox committed:
Merge branch 'main' of https://github.com/open-telemetry/opentelemetry-python-contrib into cleanup/remove-pylint-disable-celery

forafox commented:
> If we are going to move this globally I would expect the other 30 occurences of the pylint disable to move removed too. Said that the lint warning may be useful in other cases.

 
Pull request updated
@xrmx

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

forafox committed:
tests: format logging instrumentor fixtures

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

tammy-baylis-swi reviewed (approved):
Thanks, lgtm! Others will also review

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery and add CHANGELOG entry

forafox committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

tammy-baylis-swi reviewed (approved):
Lgtm again, a Maintainer (not me) will have to merge. Please resolve the latest merge conflict.

forafox committed:
Merge upstream main into cleanup/remove-pylint-disable-celery

forafox committed:
Merge remote-tracking branch 'origin/cleanup/remove-pylint-disable-celery' into cleanup/remove-pylint-disable-celery

forafox committed:
fix: resolve ruff linting errors after merge

- Add missing `Instrument` import in botocore __init__.py
- Fix PLC0207 string split in boto3sqs __init__.py

forafox committed:
fix: apply ruff format to botocore __init__.py

MikeGoldsmith reviewed (approved):
Thanks @forafox 🚀

aabmass committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

xrmx committed:
Update CHANGELOG.md

lzchen committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery

lzchen committed:
Merge branch 'main' into cleanup/remove-pylint-disable-celery
